sasl.go

```diff
@@ -216,7 +216,7 @@ func (s saslXOAUTH2Handler) step() stateFunc {
 			return nil
 		}
 	default:
-		s.conn.err = fmt.Errorf("unexpected frame type %T", fr.Body)
+		s.conn.err = fmt.Errorf("sasl: unexpected frame type %T", fr.Body)
```

Comment:
Just being thorough? Or were you able to get this to happen?

Reply:
Not yet as I haven't wrote unit tests for this.  Mostly there were too many generic "unexpected frame type" logs, so the context of where that happened was lost.